src/main/java/dk/aau/cs/model/tapn/TimedArcPetriNetNetwork.java

```diff
@@ -686,7 +686,7 @@ public Integer getVariableIndex(String name) {
 	
 	public boolean isIndeticalToExisting(ColorType newColorType) {
 		for (ColorType ct : colorTypes) {
-			if (ct.getSemanticValue().equals(newColorType.getSemanticValue()) && !ct.equals(newColorType)) {
+			if (ct.isIdentical(newColorType) && !ct.equals(newColorType)) {
```

Comment:
nice, I think you could also add the equals check to the isIdentical, so comparing with once self is also handled there :)